tlslite/tlsrecordlayer.py

```diff
@@ -315,6 +316,10 @@ def readAsync(self, max=None, min=1):
                         result.time = time.time()
                         self.tickets.append(result)
                         continue
+                    if isinstance(result, KeyUpdate):
+                        for _ in self._handle_keyupdate_request(result):
+                            pass
```

Comment:
it should actually yield result here too - when the writing is blocking it will hang here